### PR TITLE
Clean up unused Enum localization code

### PR DESCRIPTION
**Pull request by lesaltzm**
This is a clean up of some enum code that has not been used since PowerApps' Beta. We get a bunch of benefits besides removing debt from this, it significantly reduces the number of resources we need to load on startup, as well as shrinks the resource package size a bit

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStoreBuilder.cs [hunk 1]
```diff
@@ -188,7 +188,7 @@ internal sealed class EnumStoreBuilder
                 },
                 {
                     EnumConstants.ErrorKindEnumString,
-                    "%n[None:0, Sync:1, MissingRequired:2, CreatePermission:3, EditPermission:4, DeletePermission:5, Conflict:6, NotFound:7, " +
+                    "%n[None:0, Sync:1, MissingRequired:2, CreatePermission:3, EditPermissions:4, DeletePermissions:5, Conflict:6, NotFound:7, " +
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStoreBuilder.cs:191]: >EditPermissions:4, DeletePermissions:5 [](http://example.com/codeflow?start=79&length=38)

Good catch

File: src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumSymbol.cs [hunk 2]
```diff
@@ -12,133 +14,34 @@ namespace Microsoft.PowerFx.Core.Types.Enums
...
+        public EnumSymbol(DName name, DType enumSpec)
         {
-            Contracts.AssertValid(invariantName);
-            Contracts.Assert(invariantType.IsEnum);
+            Contracts.AssertValid(name);
+            Contracts.Assert(enumSpec.IsEnum);
 
             Name = name;
-            InvariantName = invariantName;
-            EnumType = invariantType;
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumSymbol.cs:37]: >EnumType [](http://example.com/codeflow?start=12&length=8)

Love seeing all this red

**Review by CarlosFigueiraMSFT — DISMISSED** [review]
:shipit:

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.IsInternal(Microsoft.OpenApi.Models.OpenApiParameter)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.ToFormulaType(Microsoft.OpenApi.Models.OpenApiSchema)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.TryGetDefaultValue(Microsoft.OpenApi.Models.OpenApiSchema,System.String@)

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Binding/INameResolver.cs [hunk 3]
```diff
@@ -58,12 +58,6 @@ internal interface INameResolver
         /// </returns>
         IEnumerable<TexlFunction> LookupFunctionsInNamespace(DPath nameSpace);
 
-        // Return true if the specified boxed enum info contains a value for the given locale-specific name.
-        bool LookupEnumValueByInfoAndLocName(object enumInfo, DName locName, out object value);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Binding/INameResolver.cs:62]: Yes!!!

**Review by MikeStall — APPROVED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs [hunk 4]
```diff
@@ -112,12 +115,11 @@ public static void LogTelemetryForFunction(TexlFunction function, CallNode node,
         /// <returns>
         /// True if a valid overload was found, false if not.
         /// </returns>
-        internal static bool TryGetBestOverload(CheckTypesContext context, IErrorContainer errors, CallNode node, DType[] argTypes, TexlFunction[] overloads, out TexlFunction bestOverload, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap, out DType returnType)
+        internal static bool TryGetBestOverload(CheckTypesContext context, IErrorContainer errors, CallNode node, TexlNode[] args, DType[] argTypes, TexlFunction[] overloads, out TexlFunction bestOverload, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap, out DType returnType)
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs:118]: >TexlNode[] args, [](http://example.com/codeflow?start=114&length=16)

Why are we changing this?

File: src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStoreBuilder.cs [hunk 5]
```diff
@@ -188,7 +188,7 @@ internal sealed class EnumStoreBuilder
...
                     "ConstraintViolated:8, GeneratedValue:9, ReadOnlyValue:10, Validation: 11, Unknown: 12, Div0: 13, BadLanguageCode: 14, " +
                     "BadRegex: 15, InvalidFunctionUsage: 16, FileNotFound: 17, AnalysisError: 18, ReadPermission: 19, NotSupported: 20, " +
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStoreBuilder.cs:193]: >ReadPermission [](http://example.com/codeflow?start=98&length=14)

This is probably also "ReadPermissions"?

File: src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStoreBuilder.cs (hunk 1 above)
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStoreBuilder.cs:191]: >EditPermissions:4, DeletePermissions:5 [](http://example.com/codeflow?start=79&length=38)

Why is this changed? Enum value is EditPermission.

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs (hunk 4 above)
Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs:118]: Small unrelated change that was found while doing the PA Client buddy build, hits a debug assert in dataflow analysis for string interp calls. Easy/safe  enough to fix while I was in there, so I just went ahead and did it.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.
